modules/fedimint-gw-client/src/pay.rs

```diff
@@ -0,0 +1,17 @@
+/// State machine that pays a lightning invoice on behalf of a federation user.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    PayInvoice -- validate outgoing contract --> Canceled
+///    PayInvoice -- await preimage offer accepted --> BuyPreimage
+///    PayInvoice -- await preimage offer failure --> Canceled
+///    BuyPreimage -- await federation decryption --> Preimage
+///    BuyPreimage -- external payment --> Preimage
```

Comment:
This might deserve a step in-between to commit to which branch we are taking (internal/external) to prevent that after a restart the list of available GWs changed and we'd choose differently the second time around (potentially trying to buy the same preimage twice in two different ways).

Reply:
Done